t/scripts/run_timeout.py

```diff
@@ -1,12 +1,25 @@
 #!/usr/bin/env python3
 """run command with a timeout, timeout as a float is first argument, rest are command"""
 import sys
+import signal
 import subprocess as s
 
-if len(sys.argv) < 2:
+if len(sys.argv) < 3:
     print("insufficient arguments passed to run_timeout")
-TIMEOUT = float(sys.argv[1])
+    sys.exit(1)
+SIG = signal.SIGKILL
+ARG_COUNT = 1
+if sys.argv[1] == "-a":
+    SIG = signal.SIGALRM
+    ARG_COUNT = 2
+    if len(sys.argv) < 4:
+        print("no command specified")
+        sys.exit(1)
+TIMEOUT = float(sys.argv[ARG_COUNT])
 try:
-    sys.exit(s.run(sys.argv[2:], timeout=TIMEOUT).returncode)
+    P = s.Popen(sys.argv[ARG_COUNT + 1 :])
+    sys.exit(P.wait(timeout=TIMEOUT))
 except s.TimeoutExpired:
-    sys.exit(142)
+    P.send_signal(SIG)
+    r = P.wait()
```

Comment:
I'm not sure how important this is, but if `SIG == signal.SIGALRM`, do we want to wait for a finite length and then follow up with `SIGKILL`?

Reply:
Good idea, added.